presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1567,6 +1584,86 @@ public PhysicalOperation visitJoin(JoinNode node, LocalExecutionPlanContext cont
             }
         }
 
+        private PhysicalOperation createSpatialJoin(JoinNode node, LocalExecutionPlanContext context)
+        {
+            Verify.verify(node.getFilter().isPresent());
+            Expression filterExpression = node.getFilter().get();
+            List<FunctionCall> spatialFunctions = extractSupportedSpatialFunctions(filterExpression);
+            for (FunctionCall spatialFunction : spatialFunctions) {
+                List<Expression> arguments = spatialFunction.getArguments();
```

Comment:
what if `spatialFunction` is used within `NOT`? As I understand spatial join will return wrong records.

Reply:
Thanks